test/util.sh

```diff
@@ -0,0 +1,203 @@
+#!/bin/bash
+# This utility file contains functions that wrap commands to be tested and allow us to interface 
+# our command-line tests with JUnit on Jenkins. All wrapper functions run commands in a sub-shell 
+# and redirect all output. Tests in test-cmd *must* use these functions for testing.
+
+# HIGH-LEVEL WRAPPER FUNCTIONS: ASSUMING EXIT CODES
+
+# expect_success runs the cmd and expects an exit code of 0
+function os::test::expect_success() {
+	cmd=$1
+
+	os::test::expect_exit_code_run_grep "${cmd}"
+}
+
+# expect_failure runs the cmd and expects a non-zero exit code
+function os::test::expect_failure() {
+	cmd=$1
```

Comment:
Thinking about it, the most common mistake is likely to be bad escaping of arguments.  Add a check to make sure you only have one arg.